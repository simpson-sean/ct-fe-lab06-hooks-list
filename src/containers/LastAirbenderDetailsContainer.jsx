import React, { useState, useEffect } from 'react';
import { findCharacterById } from '../services/LastAirbenderAPI';
import { Link, useParams } from 'react-router-dom';
import Character from '../components/characters/Character';


const LastAirbenderDetailsContainer = () => {

    const [loading, setLoading] = useState(true);
    const [character, setCharacter] = useState({});
    const {id} = useParams()

    useEffect( () => {
        findCharacterById(id).then((character) => {
            setCharacter(character);
            setLoading(false);
        })
    }, [] )

        if(loading)return<img src="https://powerusers.microsoft.com/t5/image/serverpage/image-id/118082i204C32E01666789C?v=v2"
        alt="loading spinner" />

        return (
            <div data-testid="testing-div">
                <Character name={character.name} image={character.image} gender={character.gender} profession={character.profession} />
                <Link to={`/`}>Go Back</Link>
            </div>
        )
    }

    export default LastAirbenderDetailsContainer;