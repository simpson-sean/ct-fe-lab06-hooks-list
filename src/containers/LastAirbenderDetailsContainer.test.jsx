import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router';
import detailData from '../fixtures/detail.json';
import LastAirbenderDetailsContainer from './LastAirbenderDetailsContainer';


const server = setupServer(
  rest.get(`https://last-airbender-api.herokuapp.com/api/v1/characters/:id`, (req, res, ctx) => {
    return res(ctx.json(detailData));
  })
);

describe('LastAirbenderDetails', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('display a single character', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/5cf5679a915ecad153ab68c9']}>
        <LastAirbenderDetailsContainer />
      </MemoryRouter>
    );

    screen.getByAltText("loading spinner");

    await screen.findByText('Aang', { exact: false });
    expect(container).toMatchSnapshot();
  });
});