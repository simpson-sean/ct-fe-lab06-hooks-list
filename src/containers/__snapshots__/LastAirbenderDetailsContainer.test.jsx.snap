// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastAirbenderDetails display a single character 1`] = `
<div>
  <div
    data-testid="testing-div"
  >
    <img
      alt="Aang"
      src="https://vignette.wikia.nocookie.net/avatar/images/a/ae/Aang_at_Jasmine_Dragon.png/revision/latest?cb=20130612174003"
    />
    <h1>
      Aang
    </h1>
    <h3>
      Male
    </h3>
    <h3>
       Air Nomad culture teacher Airbending instructor Avatar Monk
    </h3>
    <a
      href="/"
    >
      Go Back
    </a>
  </div>
</div>
`;
